core/pta/rockchip/rk_secure_boot.c

```diff
@@ -0,0 +1,406 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2025, Pengutronix, Michael Tretter <entwicklung@pengutronix.de>
+ */
+
+#include <config.h>
+#include <drivers/rockchip_otp.h>
+#include <kernel/pseudo_ta.h>
+#include <kernel/tee_misc.h>
+#include <tee/uuid.h>
+#include <utee_defines.h>
+#include <stdio.h>
+#include <string.h>
+#include <platform_config.h>
+
+#include <pta_rk_secure_boot.h>
+
+#define PTA_NAME "rk_secure_boot.pta"
+
+/*
+ * Disable simulation to actually fusing the hash into the OTP. Enabling this
+ * option is necessary to actually enable secure boot with PTA, but may
+ * potentially brick your device.
+ */
+static const int simulation = IS_ENABLED2(CFG_RK_SECURE_BOOT_SIMULATION);
+
+static inline bool bit_test(uint32_t value, uint32_t bit)
```

Comment:
This function name is a potential conflict with a macro with the same name in `<bitstring.h>`.